### PR TITLE
Update developer guide

### PR DESCRIPTION
**Pull request by EuniceSim142**
Changes:

- Update Target User Profile
- Update Value Proposition
- Update Non-Functional Requirements
- Update Glossary

closes https://github.com/AY2122S1-CS2103T-W16-2/tp/issues/12

### CODE REVIEW

**Review by Imranr2 — COMMENTED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -312,16 +317,39 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+| b/ | BIRTH_DATE | Client |
+| c/ | IS_COMPLETED | Booking |
+| d/ | DATE | Booking |
+| e/ | EMAIL | Client |
+| m/ | START_DATE | Membership |
+| n/ | NAME | Client |
+| p/ | PHONE_NUMBER | Client |
+| pe/ | PERIOD | Membership |
+| s/ | START_TIME | Booking |
+| v/ | IS_VACCINATED | Client |
```
Comment (sivayogasubramanian) [inline docs/DeveloperGuide.md:352]: Minor nit with formatting to standardise with table found at line 278
```suggestion
| Prefix | Parameter     | Associated with |
| :----: | :------------ | :-------------: |
| a/     | ADDRESS       | Client          |
| b/     | BIRTH_DATE    | Client          |
| c/     | IS_COMPLETED  | Booking         |
| d/     | DATE          | Booking         |
| e/     | EMAIL         | Client          |
| m/     | START_DATE    | Membership      |
| n/     | NAME          | Client          |
| p/     | PHONE_NUMBER  | Client          |
| pe/    | PERIOD        | Membership      |
| s/     | START_TIME    | Booking         |
| v/     | IS_VACCINATED | Client          |
```
Could also help change the formatting of the table in line 278? The title row is a bit off

**Review by sivayogasubramanian — COMMENTED** [review]
Looks good other than a few minor things

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -312,16 +317,39 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
-
+1. Usability Requirements:
+    * Should work on systems with Java 11 and above
+    * Should work for colorblind individuals
```
[the remark this answers is not in the document]
Comment (zognin) [reply docs/DeveloperGuide.md:322]: @sivayogasubramanian No need for now haha

**Review by sivayogasubramanian — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**